### PR TITLE
ContentCulturePicker - Add ability to set Localization Cookie

### PR DESCRIPTION
**Pull request by jptissot**
Adds the ability to set the Localization cookie when switching locales using the ContentCulturePicker

### CODE REVIEW

**Comment by jptissot** [comment]
I will also add a setting to choose where the Controller redirects the user (either on the same page or on the homepage of the locale) because they are both conflicting.

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/ContentCulturePickerController.cs [hunk 1]
```diff
@@ -46,6 +53,18 @@ public async Task<IActionResult> RedirectToLocalizedContent(string targetCulture
             {
                 return LocalRedirect('~' + contentItemUrl);
             }
+            var setCookie = (await _siteService.GetSiteSettingsAsync()).As<ContentCulturePickerSettings>()?.SetCookie;
+
+            if(setCookie.HasValue && setCookie.Value)
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/ContentCulturePickerController.cs]: space

File: src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/ContentCulturePickerController.cs [hunk 2]
```diff
@@ -46,6 +53,18 @@ public async Task<IActionResult> RedirectToLocalizedContent(string targetCulture
...
+            {
+                // Set the cookie to handle redirecting to a custom controller
+                Response.Cookies.Delete(CookieRequestCultureProvider.DefaultCookieName);
+                Response.Cookies.Append(
+                    CookieRequestCultureProvider.DefaultCookieName,
+                    CookieRequestCultureProvider.MakeCookieValue(new RequestCulture(targetCulture)),
+                    new CookieOptions { Expires = DateTime.UtcNow.AddMonths(1) }
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/ContentCulturePickerController.cs]: Use what ASP.NET templates do for this cookie

Comment (jptissot) [reply src/OrchardCore.Modules/OrchardCore.ContentLocalization/Controllers/ContentCulturePickerController.cs]: I assume that you mean the Authentication cookie ? 
https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authentication/Cookies/src/CookieAuthenticationOptions.cs#L31

**Comment by jptissot** [comment]
I added a setting that toggles the Homepage redirect. By default it returns to the same page but with the culture changed (if the cookie is set)

**Comment by jptissot** [comment]
@agriffard Can you review this one please ?

**Comment by agriffard** [comment]
Please add a ContentLocalization menu entry in mkdocs.yml or the page will never appear in the Documentation.

**Comment by sebastienros** [comment]
Documentation please when you have some time.

**Comment by agriffard** [comment]
Documentation added in mkdocs.yml
@jptissot : Can you please complete it by adding a Liquid Example for the ContentCulturePickerContainer (It is still a TODO)

**Pull request merged**